examples/decode_progressive.cc

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This C++ example decodes a JPEG XL image progressively (input bytes are
+// passed in chunks). The example outputs the intermediate steps to PAM files.
+
+#include <inttypes.h>
+#include <limits.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <vector>
+
+#include "jxl/decode.h"
+#include "jxl/decode_cxx.h"
+#include "jxl/resizable_parallel_runner.h"
+#include "jxl/resizable_parallel_runner_cxx.h"
+
+/*
+// Example code for outputting to PNG instead (requires lodepng)
+
+#include <lodepng.h>
```

Comment:
Did you rebase after https://github.com/libjxl/libjxl/pull/923? Should here too libpng be used?

Reply:
Maybe I should just drop that big comment with a png example. The non-commented-out code outputs .pam, which is fine for ImageMagick but it's a bit silly that it's a format we don't actually support ourselves (maybe we should). I wanted something with alpha so progressive alpha can also be demo'ed with this.

Reply:
I see. Perhaps either do a libpng example or drop the example altogether, but I'm also fine with leaving the lodepng one as an example. It made a bit more sense when we still depended on it..

Reply:
I removed the commented out png encoding example, it's not very useful and it's not a good practice to have big chunks of commented out code like that.